src/sidebar/components/TopBar.js

```diff
@@ -63,110 +64,101 @@ function TopBar({
+        'absolute h-10 left-0 top-0 right-0 z-4 transform-gpu',
+        'text-grey-7 border-b theme-clean:border-b-0 bg-white'
       )}
-      {/* Sidebar view */}
-      {isSidebar && (
-        <div className="TopBar__inner content">
-          <GroupList />
-          <div className="hyp-u-stretch" />
-          {pendingUpdateCount > 0 && (
-            <IconButton
-              icon="refresh"
-              onClick={applyPendingUpdates}
-              size="small"
-              variant="primary"
-              title={`Show ${pendingUpdateCount} new/updated ${
-                pendingUpdateCount === 1 ? 'annotation' : 'annotations'
-              }`}
-            />
-          )}
-          <SearchInput
-            query={filterQuery || null}
-            onSearch={store.setFilterQuery}
-          />
-          <SortMenu />
-          {showSharePageButton && (
-            <IconButton
-              icon="share"
-              expanded={isAnnotationsPanelOpen}
-              onClick={toggleSharePanel}
-              size="small"
-              title="Share annotations on this page"
-            />
+      data-testid="top-bar"
+    >
+      <SidebarContent
+        classes={classnames(
+          'flex items-center h-full',
+          // This precise horizontal padding makes the edges of its contents
+          // align accurately with the edges of annotation cards in the sidebar
+          'px-[9px]',
+          // Text sizing will size icons in buttons correctly
+          'text-xl'
+        )}
+        data-testid="top-bar-content"
+      >
+        {isSidebar ? <GroupList /> : <StreamSearchInput />}
+        <div className="grow flex items-center justify-end">
+          {isSidebar && (
+            <>
+              {pendingUpdateCount > 0 && (
+                <IconButton
+                  icon="refresh"
+                  onClick={onApplyUpdates}
+                  size="small"
+                  variant="primary"
+                  title={`Show ${pendingUpdateCount} new/updated ${
+                    pendingUpdateCount === 1 ? 'annotation' : 'annotations'
+                  }`}
+                />
+              )}
+              <SearchInput
+                query={filterQuery || null}
+                onSearch={store.setFilterQuery}
+              />
+              <SortMenu />
+              {showSharePageButton && (
+                <IconButton
+                  icon="share"
+                  expanded={isAnnotationsPanelOpen}
+                  onClick={toggleSharePanel}
+                  size="small"
+                  title="Share annotations on this page"
+                />
+              )}
+            </>
           )}
           <IconButton
             icon="help"
             expanded={isHelpPanelOpen}
-            onClick={requestHelp}
+            onClick={onRequestHelp}
             size="small"
             title="Help"
           />
-          {loginControl}
+          {auth.status === 'logged-in' ? (
```

Comment:
I think this section could bear some refactor or possible component extraction but I'm going to leave as-is for now.